src/Sarif/Baseline/ResultMatching/HeuristicMatchers/GenericHeuristicMatcher.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching.HeuristicMatchers
+{
+    internal class GenericHeuristicMatcher : IResultMatcher
+    {
+        public GenericHeuristicMatcher(IResultMatchingComparer comparer, IRemappingCalculator remapper = null)
+        {
+            Comparer = comparer;
+            Remapper = remapper;
+        }
+
+        public IResultMatchingComparer Comparer { get; }
+        public IRemappingCalculator Remapper { get; }
+
+        public IEnumerable<MatchedResults> MatchResults(IEnumerable<MatchingResult> baseline, IEnumerable<MatchingResult> current)
+        {
+            IEnumerable<SarifLogRemapping> possibleRemappings = null;
+            Dictionary<int, List<MatchingResult>> baselineResults = new Dictionary<int, List<MatchingResult>>();
```

Comment:
> Dictionary<int, List<MatchingResult>> [](start = 11, length = 38)

this is a good place to use 'var' as the assigned type is explicit (also on the next line)